src/pyuvdata/telescopes.py

```diff
@@ -948,3 +1331,439 @@ def get_enu_antpos(self):
+        else:
+            raise ValueError(
+                "order must be one of: 'AIPS', 'CASA', or a "
+                'list of length 4 containing only "x", "y", "r", or "l".'
+            )
+
+        for idx in range(self.Nants):
+            feed_a, feed_b = self.feed_array[idx]
+            if order.get(feed_a, 999999) > order.get(feed_b, 999999):
+                self.feed_array[idx] = self.feed_array[idx, ::-1]
+                self.feed_angle[idx] = self.feed_angle[idx, ::-1]
+
+        if run_check:
+            self.check(
+                check_extra=check_extra, run_check_acceptability=run_check_acceptability
+            )
+
+    def reorder_antennas(
+        self,
+        order="number",
+        *,
+        run_check=True,
+        check_extra=True,
+        run_check_acceptability=True,
+    ):
+        """
+        Arrange the antenna axis according to desired order.
+
+        Parameters
+        ----------
+        order: str or array like of int
+            If a string, allowed values are "name" and "number" to sort on the antenna
+            name or number respectively. A '-' can be prepended to signify descending
+            order instead of the default ascending order (e.g. "-number"). An array of
+            integers of length Nants representing indexes along the existing `ant_array`
+            can also be supplied to sort in any desired order (note these are indices
+            into the `ant_array` not antenna numbers).
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        ValueError
+            Raised if order is not an allowed string or is an array that does not
+            contain all the required numbers.
+
+        """
+        if isinstance(order, np.ndarray | list | tuple):
+            if not np.array_equal(np.sort(order), np.arange(self.Nants)):
+                raise ValueError(
+                    "If order is an index array, it must contain all indices for the"
+                    "ant_array, without duplicates."
+                )
+            index_array = order
+        else:
+            if order not in ["number", "name", "-number", "-name"]:
+                raise ValueError(
+                    "order must be one of 'number', 'name', '-number', '-name' or an "
+                    "index array of length Nants_data"
+                )
+
+            index_array = np.argsort(
+                self.antenna_names if "name" in order else self.antenna_numbers
+            )
+
+            if order[0] == "-":
+                index_array = np.flip(index_array)
+
+        self._select_along_param_axis({"Nants": index_array})
+
+        if run_check:
+            self.check(
+                check_extra=check_extra, run_check_acceptability=run_check_acceptability
+            )
+
+    def __add__(
+        self,
+        other: Telescope,
+        *,
+        warning_params: list | None = None,
```

Comment:
This parameter is not in the docstring here or in `__iadd__`

Reply:
Added in 1b8340609b362fc9b9225a3eb7e9535b7793952d.